File: app/src/main/java/com/lucagiorgetti/surprix/ui/mainfragments/missingowners/MissingOwnersAdapter.java
package com.lucagiorgetti.surprix.ui.mainfragments.missingowners;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.lucagiorgetti.surprix.R;
import com.lucagiorgetti.surprix.model.User;

import java.util.ArrayList;
import java.util.List;

/**
 * Adapter for showing a list of User which owns a selected surprise.
 * <p>
 * Created by Luca on 28/10/2017.
 */

public class MissingOwnersAdapter extends RecyclerView.Adapter<MissingOwnersAdapter.ViewHolder> {

    private List<User> items = new ArrayList<>();
    private MissingOwnersFragment.MyClickListener listener;

    void setOwners(List<User> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.missing_owners_element, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        User user = items.get(position);
        holder.vUsername.setText(user.getUsername());
        holder.vCountry.setText(user.getCountry());

        holder.vCard.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                listener.onOwnerClicked(user);
            }
        });
    }

    @Override
    public int getItemCount() {
        if (items == null) {
            return 0;
        }
        return items.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        TextView vUsername;
        TextView vCountry;
        View vCard;

        ViewHolder(View v) {
            super(v);
            vUsername = v.findViewById(R.id.txv_collector_username);
            vCountry = v.findViewById(R.id.txv_collector_country);
            vCard = v.findViewById(R.id.missing_owner_card);
        }
    }

    public void setListener(MissingOwnersFragment.MyClickListener myClickListener) {
        this.listener = myClickListener;
    }


}

File: app/src/main/java/com/lucagiorgetti/surprix/ui/mainfragments/missingowners/MissingOwnersAdapter.kt
package com.lucagiorgetti.surprix.ui.mainfragments.missingowners

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.lucagiorgetti.surprix.R
import com.lucagiorgetti.surprix.model.User

/**
 * Adapter for showing a list of User which owns a selected surprise.
 *
 *
 * Created by Luca on 28/10/2017.
 */
class MissingOwnersAdapter : RecyclerView.Adapter<MissingOwnersAdapter.ViewHolder>() {
    private var items: List<User?>? = ArrayList()
    private var listener: MissingOwnersFragment.MyClickListener? = null
    fun setOwners(items: List<User?>?) {
        this.items = items
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.element_missing_owners, parent, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val user = items!![position]!!
        holder.vUsername.text = user.username
        holder.vCountry.text = user.country
        holder.vCard.setOnClickListener { listener!!.onOwnerClicked(user) }
    }

    override fun getItemCount(): Int {
        return if (items == null) {
            0
        } else items!!.size
    }

    class ViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        var vUsername: TextView
        var vCountry: TextView
        var vCard: View

        init {
            vUsername = v.findViewById(R.id.txv_collector_username)
            vCountry = v.findViewById(R.id.txv_collector_country)
            vCard = v.findViewById(R.id.missing_owner_card)
        }
    }

    fun setListener(myClickListener: MissingOwnersFragment.MyClickListener?) {
        listener = myClickListener
    }
}
